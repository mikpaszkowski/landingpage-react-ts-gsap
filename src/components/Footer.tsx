import * as React from "react";
import { Logo } from "./Logo";
import styled from "styled-components";
import { colors, fontTypes } from "../styles/styleConstants";

const FooterWrapper = styled.div`
  position: relative;
  width: 100%;
  padding: 8.9rem 16.5rem 4.7rem 16.5rem;
  background-color: ${colors.footerBackground};
`;

const BreakLine = styled.div`
  height: 0.1rem;
  background-color: white;
  width: 100%;
  opacity: 0.2;
`;

const TermsAndConditions = styled.p`
  position: absolute;
  bottom: 1.6rem;
  right: 16.6rem;
  font-family: ${fontTypes.secondaryFont};
  color: ${colors.primaryLight};
  opacity: 0.5;
`;

export const Footer = () => {
  return (
    <FooterWrapper>
      <Logo inverse />
      <BreakLine />
      <TermsAndConditions>Therms & Conditions</TermsAndConditions>
    </FooterWrapper>
  );
};
