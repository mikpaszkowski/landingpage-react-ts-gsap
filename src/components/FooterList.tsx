import * as React from "react";
import styled from "styled-components";
import { colors, fontTypes } from "../styles/styleConstants";
import { lists } from "../assets/footerLists/footerLists";

const List = styled.ul`
  & > li {
    color: ${colors.primaryLight};
    font-family: ${fontTypes.secondaryFont};
    font-style: normal;
    font-weight: normal;
    font-size: 1.8rem;
    line-height: 2.7rem;
    list-style-type: none;

    &:first-child {
      color: ${colors.secondaryLight};
      margin-bottom: 1.9rem;
    }
  }
`;

interface FooterListProps {
  listName: string;
}

export const FooterList = ({ listName }: FooterListProps) => {
  return (
    <List>
      <li>{listName}</li>
      {lists[`${listName.toLocaleLowerCase()}`].map((item) => (
        <li>{item}</li>
      ))}
    </List>
  );
};
