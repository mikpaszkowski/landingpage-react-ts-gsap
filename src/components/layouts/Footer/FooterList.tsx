import * as React from "react";
import styled from "styled-components";
import { fontTypes } from "../../../styles/constants";
import { lists } from "../../staticContents/footer/footerLists";
import { device } from "../../../styles/responsive";

const List = styled.ul`
  & > li {
    color: ${({ theme }) => theme.palette.primary.fontLight};
    font-family: ${fontTypes.secondaryFont};
    font-style: normal;
    font-weight: normal;
    font-size: 1.8rem;
    line-height: 2.7rem;
    list-style-type: none;

    &:first-child {
      color: ${({ theme }) => theme.palette.secondary.fontLight};
      margin-bottom: 1.9rem;
    }
  }

  @media ${device.mobileMedium} {
    margin-bottom: 3.1rem;
  }
`;

interface FooterListProps {
  listName: string;
}

export const FooterList = ({ listName }: FooterListProps) => {
  return (
    <List>
      <li>{listName}</li>
      {lists[`${listName.toLocaleLowerCase()}`].map((item, index) => (
        <li key={index}>{item}</li>
      ))}
    </List>
  );
};
