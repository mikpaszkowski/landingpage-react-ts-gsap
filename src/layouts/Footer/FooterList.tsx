import * as React from "react";
import styled from "styled-components";
import { fontTypes } from "../../styles/constants";
import { lists } from "../../data/staticContents/footer/footerLists";
import { device } from "../../styles/responsive";

const List = styled.ul`
  margin-bottom: 3.1rem;
  & > li {
    font-family: ${fontTypes.secondaryFont};
    font-style: normal;
    font-weight: normal;
    font-size: 1.8rem;
    line-height: 2.7rem;
    list-style-type: none;

    & a {
      color: ${({ theme }) => theme.palette.primary.fontLight};
      text-decoration: none;
      font-size: 1.8rem;
      line-height: 2.7rem;
    }

    &:first-child {
      text-transform: uppercase;
      color: ${({ theme }) => theme.palette.secondary.fontLight};
      margin-bottom: 1.9rem;
    }
  }

  @media ${device.mobileMedium} {
    margin-bottom: 3.1rem;
  }
`;

interface IFooterListProps {
  listName: string;
}

export const FooterList = ({ listName }: IFooterListProps) => {
  return (
    <List>
      <li>{listName}</li>
      {lists[`${listName.toLocaleLowerCase()}`].map(({ url, label }, index) => (
        <li key={index}>
          <a href={url}>{label}</a>
        </li>
      ))}
    </List>
  );
};
