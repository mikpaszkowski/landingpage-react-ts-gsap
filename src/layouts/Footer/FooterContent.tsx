import * as React from "react";
import styled from "styled-components";
import { FooterList } from "./FooterList";
import { EmailInputContainer } from "../../components/EmailInputContainer";
import { device } from "../../styles/responsive";

const FooterContentWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: flex-start;
  margin: 5.1rem 0 8rem 0;

  @media ${device.tabletLarge} {
    flex-direction: column-reverse;
    justify-content: flex-start;
    margin: 5.6rem 0 0 0;
  }
`;

const FooterListWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: flex-start;

  ul:nth-child(2) {
    margin-right: 14rem;
  }

  ul:first-child {
    margin-right: 5.8rem;
  }

  @media ${device.tabletLarge} {
    flex-direction: column;
    justify-content: flex-start;
  }
`;

const footerLists = [
  {
    id: 1,
    listName: "products",
  },
  {
    id: 2,
    listName: "company",
  },
  {
    id: 3,
    listName: "help",
  },
];

export const FooterContent = () => {
  return (
    <FooterContentWrapper>
      <FooterListWrapper>
        {footerLists.map(({ id, listName }) => (
          <FooterList key={id} listName={listName} />
        ))}
      </FooterListWrapper>
      <EmailInputContainer />
    </FooterContentWrapper>
  );
};
