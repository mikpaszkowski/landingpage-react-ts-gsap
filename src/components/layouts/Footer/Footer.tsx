import * as React from "react";
import { Logo } from "../../ReusableComponents/Logo/Logo";
import styled from "styled-components";
import { fontTypes } from "../../../styles/styleConstants";
import { FooterContent } from "./FooterContent";
import { device } from "../../../styles/responsive";

const FooterWrapper = styled.div`
  position: relative;
  width: 100%;
  padding: 8.9rem 16.5rem 4.7rem 16.5rem;
  background-color: ${({ theme }) => theme.palette.primary.footerBackground};

  @media ${device.mobileMedium} {
    padding: 6.4rem 2.4rem 5.5rem 2.4rem;
  }
`;

const BreakLine = styled.div`
  height: 0.1rem;
  background-color: white;
  width: 100%;
  opacity: 0.2;

  @media ${device.mobileMedium} {
    display: none;
  }
`;

const TermsAndConditions = styled.p`
  position: absolute;
  bottom: 1.6rem;
  right: 16.6rem;
  font-family: ${fontTypes.secondaryFont};
  color: ${({ theme }) => theme.palette.primary.fontLight};
  opacity: 0.5;

  @media ${device.mobileMedium} {
    display: none;
  }
`;

export const Footer = () => {
  return (
    <FooterWrapper>
      <Logo inverse />
      <FooterContent />
      <BreakLine />
      <TermsAndConditions>Therms & Conditions</TermsAndConditions>
    </FooterWrapper>
  );
};
