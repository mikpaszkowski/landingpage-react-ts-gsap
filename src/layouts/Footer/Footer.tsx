import * as React from "react";
import { Logo } from "../../components/Logo/Logo";
import styled from "styled-components";
import { fontTypes } from "../../styles/constants";
import { FooterContent } from "./FooterContent";
import { device } from "../../styles/responsive";

const FooterWrapper = styled.div`
  position: relative;
  width: 100%;
  padding: 8.9rem 16.5rem 4.7rem 16.5rem;
  background-color: ${({ theme }) => theme.palette.primary.footerBackground};

  @media ${device.tablet} {
    padding: 6.4rem 2.4rem 5.5rem 2.4rem;
  }
`;

const BreakLine = styled.div`
  display: block;
  height: 0.1rem;
  background-color: white;
  width: 100%;
  opacity: 0.2;

  @media ${device.mobileLarge} {
    display: none;
  }
`;

const TermsAndConditions = styled.p`
  display: block;
  position: absolute;
  bottom: 1.6rem;
  right: 16.6rem;
  font-family: ${fontTypes.secondaryFont};
  color: ${({ theme }) => theme.palette.primary.fontLight};
  opacity: 0.5;

  @media ${device.mobileLarge} {
    display: none;
  }

  @media ${device.tablet} {
    right: 2.5rem;
  }
`;

export const Footer = () => {
  return (
    <FooterWrapper>
      <Logo inverse />
      <FooterContent />
      <BreakLine />
      <TermsAndConditions>Therms & Conditions</TermsAndConditions>
    </FooterWrapper>
  );
};
