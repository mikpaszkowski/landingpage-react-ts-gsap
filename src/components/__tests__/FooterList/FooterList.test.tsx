import { cleanup } from "@testing-library/react";
import "jest-styled-components";
import { renderWithLightTheme, mountWithDarkTheme } from "../helpers/index";
import { FooterList } from "../../layouts/Footer/FooterList";

afterEach(() => {
  cleanup();
});

test("check render correctly <FooterList /> component", () => {
  const tree = renderWithLightTheme(<FooterList listName="Company" />).toJSON();
  expect(tree).toMatchSnapshot();
});
