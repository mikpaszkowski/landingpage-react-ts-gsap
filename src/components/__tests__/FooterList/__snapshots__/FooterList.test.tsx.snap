// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check render correctly <FooterList /> component 1`] = `
.c0 > li {
  color: #E5E5E5;
  font-family: 'Reem Kufi',sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 1.8rem;
  line-height: 2.7rem;
  list-style-type: none;
}

.c0 > li:first-child {
  color: #7C8176;
  margin-bottom: 1.9rem;
}

@media (max-width:375px) {
  .c0 {
    margin-bottom: 3.1rem;
  }
}

<ul
  className="c0"
>
  <li>
    Company
  </li>
  <li>
    About
  </li>
  <li>
    Blog
  </li>
  <li>
    Join Us
  </li>
</ul>
`;
